libkbfs/journal_server.go

```diff
@@ -176,18 +218,130 @@ func (j *JournalServer) Enable(ctx context.Context, tlfID TlfID) (err error) {
 		return err
 	}
 
-	bundle.blockJournal = blockJournal
 	mdJournal, err := makeMDJournal(
 		uid, key, j.config.Codec(), j.config.Crypto(), tlfDir, log)
 	if err != nil {
 		return err
 	}
 
-	bundle.mdJournal = mdJournal
+	bundle := makeTlfJournalBundle(blockJournal, mdJournal)
 	j.tlfBundles[tlfID] = bundle
+
+	go j.autoFlush(tlfID, afs, bundle.hasWorkCh, bundle.pauseCh,
+		bundle.resumeCh, bundle.shutdownCh)
+
+	// Signal work to pick up any existing journal entries.
+	select {
+	case bundle.hasWorkCh <- struct{}{}:
+	default:
+	}
+
 	return nil
 }
 
+func (j *JournalServer) autoFlush(
```

Comment:
It seems to me like this operates entirely on fields of `tlfJournalBundle`. It might be cleaner to have this method's receiver be `tlfJournalBundle`.